src/libraries/Common/tests/System/Net/Http/ResponseStreamTest.cs

```diff
@@ -196,16 +196,29 @@ await client.GetAsync(remoteServer.EchoUri, HttpCompletionOption.ResponseHeaders
                 // Verify that the task completed.
                 Assert.True(((IAsyncResult)task).AsyncWaitHandle.WaitOne(new TimeSpan(0, 5, 0)));
                 Assert.True(task.IsCompleted, "Task was not yet completed");
-                
-                if (task.IsFaulted)
+
+                // Verify that the task completed successfully or is canceled.
+                if (IsWinHttpHandler)
                 {
-                    // Propagate exception for debugging
-                    task.GetAwaiter().GetResult();
+                    // With WinHttpHandler, we may fault because canceling the task destroys the request handle
+                    // which may randomly cause an ObjectDisposedException (or other exception).
```

Comment:
Seeing an `ObjectDisposedException` upon cancellation seems like a bug. @davidsh do you know anything about this behavior?

Reply:
Note that this is just taken from before WinHttpHandler removel in [this commit](https://github.com/dotnet/runtime/commit/ad8115d57708c491517b655bbdc0cbb619dda383#diff-843a3e2d5a9d5e9535c8deab8b47bcd7). As is every other condition returned to the test.